src/rpc/server.cpp

```diff
@@ -169,11 +170,39 @@ UniValue stop(const JSONRPCRequest& jsonRequest)
             }.ToString());
     // Event loop will exit after current HTTP requests have been handled, so
     // this reply will get back to the client.
-    StartShutdown();
-    if (jsonRequest.params[0].isNum()) {
-        MilliSleep(jsonRequest.params[0].get_int());
+    if(rpcallowstop)
+    {
+        StartShutdown();
+        if (jsonRequest.params[0].isNum()) {
+            MilliSleep(jsonRequest.params[0].get_int());
+        }
+        return "Bitcoin server stopping";
+    }
+    else
+    {
+        // Check if IP is 127.0.0.1
+        std::string ip;
+        for(int i = 0; i < jsonRequest.peerAddr.length(); i++)
+        {
+            if(jsonRequest.peerAddr[i] == ':')
+            {
+                break;
+            }
+            ip += jsonRequest.peerAddr[i];
+        }
+        if(ip == "127.0.0.1")
+        {
+            StartShutdown();
```

Comment:
This block is redundant with the one at L174 and could be extracted.

Reply:
Are you sure about this?
This block should only be executed if remote stop is disabled so a separate string slicing even if it isn't necessary is a bit too much overhead imo.